### PR TITLE
make sink operator optional

### PR DESCRIPTION
**Pull request by cooldome**
Usability improvement, users can now provide only `=destroy`, `=`. The `=sink` can be generated.

### CODE REVIEW

**Comment by cooldome** [comment]
This PR is affected by the bug #13149. This how it was discovered.
Otherwise it is ready

**Comment by cooldome** [comment]
ready for review

File: doc/destructors.rst [hunk 1]
```diff
@@ -130,7 +131,10 @@ A `=sink` hook moves an object around, the resources are stolen from the source
 and passed to the destination. It is ensured that source's destructor does
 not free the resources afterwards by setting the object to its default value
 (the value the object's state started in). Setting an object ``x`` back to its
-default value is written as ``wasMoved(x)``.
+default value is written as ``wasMoved(x)``. When not provided compiler
```
Comment (Araq) [inline doc/destructors.rst]: "the compiler"

File: doc/destructors.rst [hunk 2]
```diff
@@ -130,7 +131,10 @@ A `=sink` hook moves an object around, the resources are stolen from the source
...
+is using a combination of `=destroy` and `copyMem` instead. This is a efficient
```
Comment (Araq) [inline doc/destructors.rst]: "This is efficient"

Comment (cooldome) [reply doc/destructors.rst]: fixed

File: doc/destructors.rst (hunk 1 above)
Comment (cooldome) [reply doc/destructors.rst]: fixed

**Comment by Araq** [comment]
Ping @Clyybber, awaits your review.

**Pull request merged**